### ISSUE TITLE
with statement in call causes UnboundLocalError

### ISSUE THREAD

**Issue by dplepage** [pallets/jinja#118]
I get `UnboundLocalError: local variable 'l_x' referenced before assignment` whenever I shadow a global via a `with` block inside a `call`:

``` jinja
{% set x = 1 %}
{% call foo() %}
    {% with x = 2 %}
    {% endwith %}
{% endcall %}
```

If I remove the first line, the problem goes away.

I'm using Jinja 2.6 and Python 2.6.7.

**Comment by hivre** [comment]
most likely a duplicate of #278

**Comment by mitsuhiko** [comment]
Fixed by #637
